tools/elastic/migration/MigrationService.cs

```diff
@@ -77,29 +78,40 @@ public async Task RunAsync(CancellationToken cancellationToken)
     private async Task CreateMigrationIndexAsync(CancellationToken cancellationToken)
     {
         // Check if migration index exists.
-        var exists = await _elasticClient.Indices.ExistsAsync(Indices.Index(_options.MigrationIndex), null, cancellationToken);
-
-        // If the index does not exist, create it.
-        if (!exists.Exists)
+        try
         {
-            var model = new MigrationVersion();
-            var descriptor = new CreateIndexDescriptor(_options.MigrationIndex)
-                .Map(tmd => tmd.AutoMap());
-
-            var createResponse = await _elasticClient.Indices.CreateAsync(
-                _options.MigrationIndex,
-                cir => cir.Map<MigrationVersion>(m => m.AutoMap())
-                    .Settings(s => s
-                        .NumberOfReplicas(_options.NumberOfReplicas)
-                        .NumberOfShards(_options.NumberOfShards)),
-                cancellationToken);
-
-            if (!createResponse.IsValid)
+            var exists = await _elasticClient.Indices.ExistsAsync(Indices.Index(_options.MigrationIndex), ied => ied.ErrorTrace(true), cancellationToken);
+
+            // If the index does not exist, create it.
+            if (!exists.Exists)
             {
-                _logger.LogError(createResponse.OriginalException, "Failed to create migration index.");
-                throw createResponse.OriginalException;
+                var model = new MigrationVersion();
+                var descriptor = new CreateIndexDescriptor(_options.MigrationIndex)
+                    .Map(tmd => tmd.AutoMap());
+
+                var createResponse = await _elasticClient.Indices.CreateAsync(
+                    _options.MigrationIndex,
+                    cir => cir.Map<MigrationVersion>(m => m.AutoMap())
+                        .Settings(s => s
+                            .NumberOfReplicas(_options.NumberOfReplicas)
+                            .NumberOfShards(_options.NumberOfShards)),
+                    cancellationToken);
+
+                if (!createResponse.IsValid)
+                {
+                    _logger.LogError(createResponse.OriginalException, "Failed to create migration index.");
+                    throw createResponse.OriginalException;
+                }
             }
         }
+        catch (ElasticsearchClientException ex)
+        {
+            _logger.LogError(ex, "Failure to create index");
+            if (ex.Response.HttpStatusCode == 502)
+                await CreateMigrationIndexAsync(cancellationToken);
```

Comment:
Infinite loop until successful... Horrible implementation, but the gateway is having issues.